.gitlab-ci.yml

```diff
@@ -7,6 +7,7 @@ include:
   - "projects/vdk-control-cli/.gitlab-ci.yml"
   - "projects/vdk-core/.gitlab-ci.yml"
   - "projects/frontend/cicd/.gitlab-ci.yml"
+  - "projects/frontend/shared-components/cicd/.gitlab-ci.yml"
```

Comment:
please mind, by design (VEP-1507) the frontend subproject are handled by a common `projects/frontend/cicd/` dir (see #L9)

Reply:
We're treating vdk/shared as a separate package. Separate package means separate CI config, imo. We can always change the VEP if we decide this is a better approach.

Reply:
One Gitlab CI YAML file allows multiple release stage jobs. If in readable length, it is rather simple having a CI/CD per component, than per artifact. An example benefit is, reusing commons like change locations preferable to code duplication. Also, data-pipelines artifact build job dependsmon shared-components build job.

Reply:
I spent some time extracting the code 
https://github.com/vmware/versatile-data-kit/pull/1685

(we don't need circular dependencies, do we :)) meaning waiting on https://github.com/vmware/versatile-data-kit/pull/1671 ...)

Reply:
In fact, I am starting to think on speeding up the  change #1671 depends on, by suggesting to split
* the build & tests
* the publish
* the versioning work (this may take longer)

@DeltaMichael 
What do you think?

Reply:
We're treating vdk/shared as a separate package and it has a simpler process around it. We can add versioning for shared, as we already agreed to use the pipeline id and it's a 4-line script. I'll pull the latest changes from #1685 and go for there. This will hopefully unblock #1671